flik/jacobian.py

```diff
@@ -1,214 +0,0 @@
-# An experimental local optimization package
-# Copyright (C) 2018 Ayers Lab <ayers@mcmaster.ca>.
-#
-# This file is part of Flik.
-#
-# Flik is free software; you can redistribute it and/or
-# modify it under the terms of the GNU General Public License
-# as published by the Free Software Foundation; either version 3
-# of the License, or (at your option) any later version.
-#
-# Flik is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU General Public License for more details.
-#
-# You should have received a copy of the GNU General Public License
-# along with this program; if not, see <http://www.gnu.org/licenses/>
-
-
-r"""Base Jacobian class."""
-
-
-import numpy as np
-
-
-__all__ = [
-    "Jacobian",
-    ]
-
-
-class Jacobian:
-    r"""
-    Jacobian class with analytical evaluation by callable Jacobian function.
-
-    The Jacobian class and its subclasses are used for evaluating and updating
-    Jacobians as part of the Newton iterations.
-
-    """
-
-    def __init__(self, jac):
-        r"""
-        Construct a Jacobian class for a callable analytical jacobian.
-
-        Parameters
-        ----------
-        jac : callable, optional
-
-        Raises
-        ------
-        TypeError
-            If an argument of an invalid type or shape is passed.
-
-        """
-        if not callable(jac):
-            raise TypeError("J must be a callable object")
-        self._jac = jac
-
-    def __call__(self, x):
-        r"""
-        Compute the Jacobian at position ``x``.
-
-        Parameters
-        ----------
-        x : np.ndarray
-
-        Returns
-        -------
-        y :np.ndarray
-
-        """
-        return self._jac(x)
-
-    def update_newton(self, A, new_x, *_):
```

Comment:
Paul wants like 100 different quasi-newton updates, so I moved these to flik/methods/